addons/stock/models/stock_move.py

```diff
@@ -739,6 +741,19 @@ def _generate_serial_numbers(self, next_serial_count=False):
             ))
         move_lines_commands = self._generate_serial_move_line_commands(lot_names)
         self.write({'move_line_ids': move_lines_commands})
+        existing_lots = []
+        for lot_name in lot_names:
+            counter = self.env['stock.production.lot'].search_count([
+                ('company_id', '=', self.company_id.id),
+                ('product_id', '=', self.product_id.id),
+                ('name', '=', lot_name),
+            ])
+            if counter > 0:
+                existing_lots.append(lot_name)
+        if len(existing_lots)>0:
+            return {
+                'warning': {'title': _('Warning'), 'message': _('Existing Serial Numbers (%s). Please correct the serial numbers encoded.') % ','.join(existing_lots)}
```

Comment:
You can move the data directly inside the _() function. It ensure the '%s' is set in the translations
_('Existing Serial Numbers (%s). Please correct the serial numbers encoded.', ','.join(existing_lots))}